docs/zkapps/o1js/interact-with-mina.md

```diff
@@ -132,20 +177,33 @@ Both of these are heavy cryptographic operations and can take between a few seco
 ];
 ```
 
-You might wonder: what, exactly, is proved here? How is the proof linked to the account update it is part of?
+## How proofs link to account updates
+
+You might wonder: what, exactly, is proved? How is the proof linked to the account update it is part of?
 
 The proof attests to two different things:
 
 - The execution of `myMethod()`
 - The public input of that execution
 
-Recall that all method arguments are _private inputs_. So, the verifier doesn't get to see them, and the proof doesn't say anything about them (it only says that there were _some_ private inputs that satisfied all constraints). However, a zk proof can also have a public input. In the case of zkApps, **the public input is the account update**. It is passed in implicitly when you do `tx.prove()`. The prover function (i.e., your smart contract logic) creates its own account update and constrains it to equal the public input.
+Recall that all method arguments are _private inputs_. So, the verifier doesn't get to see them, and the proof doesn't say anything about them (it only says that there were _some_ private inputs that satisfied all constraints). However, a zk proof can also have a public input. 
+
+In the case of zkApps, **the public input is the account update** that is passed in implicitly with `tx.prove()`. The prover function (smart contract logic) creates its own account update and constrains it to equal the public input.
+
+The public input is data that is shared between the prover and verifier:
+
+- The verifier passes in the public input when verifying it.
+- The proof is valid only if it was created with _the same public input_. 
 
-You can think of the public input as data that is shared between the prover and verifier. The verifier passes in the public input when verifying it, and the proof is valid only if it was created with _the same public input_. This means that this proof attests to the validity of exactly this account update. If you change the account update before sending it to the Mina network, the proof is no longer not be valid. In other words: The only valid account updates for a zkApp account are the ones created according to the logic of your `SmartContract`. This is the core of why we can have smart contracts that execute on the client side.
+The proof attests to the validity of exactly this account update. If you change the account update before sending it to the Mina network, the proof is no longer valid. 
 
-## Payments and more on public inputs
+The only valid account updates for a zkApp account are the ones created according to the logic of your `SmartContract`. This core concept is why zkApp smart contracts execute on the client side.
```

Comment:
is there a reason to use code literal here? Is the code always `SmartContract`? or we can just use plain text?

> your `SmartContract`